### PR TITLE
feat: Adds sidebar view for unattended conversations

### PR DESCRIPTION
**Pull request by nithindavid**
## Description
Fixes - https://github.com/chatwoot/product/issues/645

 Loom video - https://www.loom.com/share/79882e47dbc6446dbd79dae42ab96ca2

**What's not done**
The view should remove the conversation from the conversation list once an agent replies to it. This could be done as an improvement

### CODE REVIEW

**Comment by muhsin-k** [comment]
@nithindavid Found a small issue while testing this PR. Please check this loom video.

https://www.loom.com/share/f1e6e388b6fd4cebb76ce7341954a0b6

File: app/finders/conversation_finder.rb [hunk 1]
```diff
@@ -79,6 +79,8 @@ def find_all_conversations
     if params[:conversation_type] == 'mention'
       conversation_ids = current_account.mentions.where(user: current_user).pluck(:conversation_id)
       @conversations = current_account.conversations.where(id: conversation_ids)
+    elsif params[:conversation_type] == 'unattended'
```
Comment (muhsin-k) [inline app/finders/conversation_finder.rb:82]: Showing **rubocop** warnings while running the app. This change in code will fix the issue.

```ruby
  def set_assignee_type
    @assignee_type = params[:assignee_type]
  end
```

```ruby
  def find_all_conversations
    case @conversation_type
    when 'mention'
      conversation_ids = current_account.mentions.where(user: current_user).pluck(:conversation_id)
      @conversations = current_account.conversations.where(id: conversation_ids)
    when 'unattended'
      @conversations = current_account.conversations.where(inbox_id: @inbox_ids, first_reply_created_at: nil)
    else
      @conversations = current_account.conversations.where(inbox_id: @inbox_ids)
    end
    @conversations
  end
```

**Comment by nithindavid** [comment]
> @nithindavid Found a small issue while testing this PR. Please check this loom video.
> 
> https://www.loom.com/share/f1e6e388b6fd4cebb76ce7341954a0b6

Thanks. Will fix it 👍

File: app/finders/conversation_finder.rb (hunk 1 above)
Comment (nithindavid) [reply app/finders/conversation_finder.rb:82]: Thanks, I have added them.

File: app/finders/conversation_finder.rb [hunk 2]
```diff
@@ -76,12 +76,16 @@ def set_team
   end
 
   def find_all_conversations
-    if params[:conversation_type] == 'mention'
+    case @conversation_type
+    when 'mention'
       conversation_ids = current_account.mentions.where(user: current_user).pluck(:conversation_id)
       @conversations = current_account.conversations.where(id: conversation_ids)
+    when 'unattended'
```
Comment (pranavrajs) [inline app/finders/conversation_finder.rb:83]: Please add a spec for the same.

File: app/javascript/dashboard/store/modules/conversations/helpers.js [hunk 3]
```diff
@@ -32,6 +33,10 @@ export const applyPageFilters = (conversation, filters) => {
     const filterByLabels = labels.every(label => chatLabels.includes(label));
     shouldFilter = shouldFilter && filterByLabels;
   }
+  if (conversationType === 'unattended') {
+    const hasNoFirstReply = !firstReplyOn;
+    shouldFilter = shouldFilter && hasNoFirstReply;
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/conversations/helpers.js:38]:
```suggestion
    shouldFilter = shouldFilter && !firstReplyOn;
```

The variable `hasNoFirstReply` is not at all useful. Please consider the comment.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
@nithindavid 

1. The conversation counts are not working properly in the mentions view

Screenshot 2022-11-21 at 6 25 14 PM

2. Please add relevant specs for the changes you had added.

**Comment by nithindavid** [comment]
@pranavrajs I have made the changes

File: app/finders/conversation_finder.rb [hunk 4]
```diff
@@ -96,8 +92,15 @@ def filter_by_assignee_type
...
+  def filter_by_conversation_type
+    case @params[:conversation_type]
+    when 'mention'
+      conversation_ids = current_account.mentions.where(user: current_user).pluck(:conversation_id)
+      @conversations = @conversations.where(id: conversation_ids)
+    when 'unattended'
+      @conversations = @conversations.where(first_reply_created_at: nil)
+    end
+    @conversations
   end
```
Comment (pranavrajs) [inline app/finders/conversation_finder.rb:104]: @nithindavid As mentioned during our chat, please remove this method and add the logic to find_all_conversations. Also, please add the specs for the same.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the individual comment.

File: app/finders/conversation_finder.rb (hunk 4 above)
Comment (nithindavid) [reply app/finders/conversation_finder.rb:104]: I have made the changes

**Review by muhsin-k — APPROVED** [review]
Tested locally.
https://www.loom.com/share/8c8d589ffac543e987d51b9599d3b044

**Pull request merged**